cookbook/src/tx_segwit-v0.md

```diff
@@ -0,0 +1,342 @@
+    let sk = SecretKey::new(&mut rand::thread_rng());
+    let pk = bitcoin::PublicKey::new(sk.public_key(secp));
+    let wpkh = pk.wpubkey_hash().expect("key is compressed");
+
+    (sk, wpkh)
+}
+```
+
+`senders_keys` generates a random private key and derives the corresponding public key hash.
+This will be useful to mock a sender.
+In a real application these would be actual secrets.
+Under the hood we are using the [`secp256k1`](https://github.com/rust-bitcoin/rust-secp256k1/) crate to generate the key pair.
+`rust-secp256k1` is a wrapper around [libsecp256k1](https://github.com/bitcoin-core/secp256k1), a C
+library implementing various cryptographic functions using the [SECG](https://www.secg.org/) curve
+[secp256k1](https://en.bitcoin.it/wiki/Secp256k1).
+We use the `SecretKey::new` method to generate a random private key `sk`.
+We then use the `PublicKey::new` method to derive the corresponding public key `pk`.
+Finally, we use the `PublicKey::wpubkey_hash` method to derive the corresponding public key hash `wpkh`.
+Note that `senders_keys` is generic over the [`Signing`](https://docs.rs/secp256k1/latest/secp256k1/trait.Signing.html) trait.
+This is used to indicate that is an instance of `Secp256k1` and can be used for signing.
+We conclude returning the private key `sk` and the public key hash `wpkh` as a tuple.
+
+```rust
+# use std::str::FromStr;
+# use bitcoin::{Address, Network};
+fn receivers_address() -> Address {
+    Address::from_str("bc1q7cyrfmck2ffu2ud3rn5l5a8yv6f0chkp0zpemf")
+        .expect("a valid address")
+        .require_network(Network::Bitcoin)
+        .expect("valid address for mainnet")
+}
+```
+
+`receivers_address` generates a receiver address.
+In a real application this would be the address of the receiver.
+We use the method `Address::from_str` to parse the string `"bc1q7cyrfmck2ffu2ud3rn5l5a8yv6f0chkp0zpemf"` into an address.
+Hence, it is necessary to import the `std::str::FromStr` trait.
+Note that `bc1q7cyrfmck2ffu2ud3rn5l5a8yv6f0chkp0zpemf` is a [Bech32](https://bitcoinops.org/en/topics/bech32/) address.
+This is an arbitrary, however valid, Bitcoin mainnet address,
+since we use the `require_network` method to ensure that the address is valid for the mainnet.
+
+```rust
+# use bitcoin::{OutPoint, ScriptBuf, TxOut, Txid, WPubkeyHash};
+# use bitcoin::hashes::Hash;
+# const DUMMY_UTXO_AMOUNT: u64 = 20_000_000;
+fn dummy_unspent_transaction_output(wpkh: &WPubkeyHash) -> (OutPoint, TxOut) {
+    let script_pubkey = ScriptBuf::new_v0_p2wpkh(wpkh);
+
+    let out_point = OutPoint {
+        txid: Txid::all_zeros(), // Obviously invalid.
+        vout: 0,
+    };
+
+    let utxo = TxOut {
+        value: DUMMY_UTXO_AMOUNT,
+        script_pubkey,
+    };
+
+    (out_point, utxo)
+}
+```
+
+`dummy_unspent_transaction_output` generates a dummy unspent transaction output (UTXO).
+This is a SegWit V0 P2WPKH (`ScriptBuf::new_v0_p2wpkh`) UTXO with a dummy invalid transaction ID (`txid: Txid::all_zeros()`),
```

Comment:
We will be testing this extensively.

We are now only testing for `bitcoin@v0.30`:

https://github.com/rust-bitcoin/www.rust-bitcoin.org/blob/b4bc52864a9144333933b800536c197dddf70fbf/cookbook/tests/Cargo.toml#L9

However, we have a dependabot routine that will update weekly the dependencies in `Cargo.toml`, which includes `bitcoin`:

https://github.com/rust-bitcoin/www.rust-bitcoin.org/blob/b4bc52864a9144333933b800536c197dddf70fbf/.github/dependabot.yml#L9-L13

So this would open a dependabot PR for updating `bitcoin` to a new version and then we would test the cookbook code against a new version of `bitcoin`

Reply:
Yeah but the function name I commented on just appears in text, its not in a rust code block. By maintenance burden I meant that, as it stands, we really have to re-read the whole cookbook each release and make sure none of the mentioned functions have changed names.